cogs/poll.py

```diff
@@ -0,0 +1,318 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) idoneam (2016-2019)
+#
+# This file is part of Canary
+#
+# Canary is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Canary is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Canary. If not, see <https://www.gnu.org/licenses/>.
+
+# discord.py requirements
+import discord
+from discord.ext import commands
+import asyncio
+
+# for the alphabet list
+import string
+
+# for the timeout
+import time
+
+ALPHABET = list(string.ascii_uppercase)
+EMOJI_ALPHABET = [
+    u"\U0001F1E6", u"\U0001F1E7", u"\U0001F1E8", u"\U0001F1E9", u"\U0001F1EA",
+    u"\U0001F1EB", u"\U0001F1EC", u"\U0001F1ED", u"\U0001F1EE", u"\U0001F1EF",
+    u"\U0001F1F0", u"\U0001F1F1", u"\U0001F1F2", u"\U0001F1F3", u"\U0001F1F4",
+    u"\U0001F1F5", u"\U0001F1F6", u"\U0001F1F7"
+]    # :regional_indicator_a: to :regional_indicator_r:
+TIME_UNITS = {

```

Comment:
Personally I don't like having different time units. What if you only accept minutes? A poll means nothing if it last only 10 seconds, and it's quite easy to convert hours -> minutes. It also makes things way easier to parse

Reply:
I agree about that a polls in seconds isn't useful, I just put it there because, why not since I had already put all the others,  but I'm not sure converting hours into minutes is something the user should have to think about.. Like if the poll timeouts after a day, it's way better to write "1 day" than "1440 minutes"